From d0d7f496473088841db91f5008ac68acf0e90b39 Mon Sep 17 00:00:00 2001
Message-Id: <d0d7f496473088841db91f5008ac68acf0e90b39.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Uma Shankar <uma.shankar@intel.com>
Date: Fri, 19 Jul 2013 11:24:07 +0530
Subject: [PATCH 036/312] FOR_UPSTREAM [VPG]: drm/i915: Optimized edp panel
 power on delays

Currrently the delays are set at 200ms.It is a very conservative value.
EDP panel can come up within 100ms. Hard coding it to 100ms for now.

(cherry picked from commit 5885d4396259cfe3ec02bca385f9aca6d267b4a3)

Issue: APDEV-1167
Change-Id: I4dd20b4b24087a1fc6c4ce7b241d4a140920bbf9
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Reviewed-on: http://android.intel.com:8080/120455
Reviewed-by: Kachhi, Naresh Kumar <naresh.kumar.kachhi@intel.com>
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Reviewed-by: Wood, Brian J <brian.j.wood@intel.com>
Tested-by: Sun, Hang L <hang.l.sun@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 3d9f8b0..98e26de 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -3934,6 +3934,13 @@ intel_dp_init_panel_power_sequencer(struct drm_device *dev,
 	intel_dp->panel_power_down_delay = get_delay(t10);
 	intel_dp->panel_power_cycle_delay = get_delay(t11_t12);
 #undef get_delay
+	/* Currrently the delays are set at 200ms.
+	 * It is a very conservative value. EDP panel can
+	 * come up within 100ms. Hard coding it to 100ms for now.
+	 * TODO : Work with IAFW team and get it programmed correctly.
+	 */
+	intel_dp->panel_power_up_delay = 100;
+	intel_dp->backlight_off_delay = 100;
 
 	DRM_DEBUG_KMS("panel power up delay %d, power down delay %d, power cycle delay %d\n",
 		      intel_dp->panel_power_up_delay, intel_dp->panel_power_down_delay,
-- 
1.7.9.5

